modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_page.ftl

```diff
@@ -172,40 +170,40 @@ public class Page<T> {
 			}
 			else if (Objects.equals(jsonParserFieldName, "facets")) {
 				if (jsonParserFieldValue != null) {
-					page.setFacets(
-						Stream.of(
-							toStrings((Object[])jsonParserFieldValue)
-						).map(
-							this::parseToMap
-						).map(
-							facets -> new Facet(
-								(String)facets.get("facetCriteria"),
-								Stream.of(
-									(Object[])facets.get("facetValues")
-								).map(
-									object -> (String)object
-								).map(
-									this::parseToMap
-								).map(
-									facetValues -> new Facet.FacetValue(Integer.valueOf((String)facetValues.get("numberOfOccurrences")), (String)facetValues.get("term"))
-								).collect(
-									Collectors.toList()
-								))
-						).collect(
-							Collectors.toList()
-						));
+					List<Facet> facets = new ArrayList<>();
+
+					for(String string : toStrings((Object[])jsonParserFieldValue)){
+						List<Facet.FacetValue> facetValues = new ArrayList<>();
+
+						Map<String, Object> jsonParserFieldValuesMap = parseToMap(string);
+
+						for(Object object : (Object[])jsonParserFieldValuesMap.get("facetValues")){
+							Map<String, Object> facetValueMap = parseToMap((String)object);
+
+							facetValues.add(
+								new Facet.FacetValue(
+									Integer.valueOf(
+										(String)facetValueMap.get(
+											"numberOfOccurrences")),
+									(String)facetValueMap.get("term")));
+						}
+
+						facets.add(new Facet((String)jsonParserFieldValuesMap.get("facetCriteria"), facetValues));
+
+					}
+
+					page.setFacets(facets);
 				}
 			}
 			else if (Objects.equals(jsonParserFieldName, "items")) {
 				if (jsonParserFieldValue != null) {
-					page.setItems(
-						Stream.of(
-							toStrings((Object[])jsonParserFieldValue)
-						).map(
-							string -> _toDTOFunction.apply(string)
-						).collect(
-							Collectors.toList()
-						));
+					List<T> items = new ArrayList<>();
```

Comment:
@jeffwu0724 Please simplify this using TransformUtil so that you can inline `items`.